### PR TITLE
add catalogNamespace into app CR

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward giantswarm/roadmap#219

## Checklist

- [x] Consider SIG UX feedback.
- [ ] Update changelog in CHANGELOG.md.
- [ ] If adding a new type, ensure that it is added to the [CRD unit tests](https://github.com/giantswarm/apiextensions/blob/d78aba91d578d56ef49f58e336035d35edc4e1b0/pkg/crd/crd_test.go#L9).

### CODE REVIEW

File: pkg/apis/application/v1alpha1/app_types.go [hunk 1]
```diff
@@ -57,6 +57,11 @@ type AppSpec struct {
 	Catalog string `json:"catalog"`
 	// +kubebuilder:validation:Optional
 	// +nullable
+	// CatalogNamespace is the namespace of the catalog this app belongs to.
```
Comment (tomahawk28) [inline pkg/apis/application/v1alpha1/app_types.go:60]: Note that this belongs to `catalog` CRs, not `appCatalog`.

**Review by rossf7 — APPROVED** [review]
LGTM

Should we add catalogNamespace to the example CR in the UT?

**Comment by rossf7** [comment]
@tomahawk28 Also, I think we should have a changelog entry for this. Since app CRs are customer-facing.

File: pkg/apis/application/v1alpha1/app_types.go (hunk 1 above)
Comment (marians) [reply pkg/apis/application/v1alpha1/app_types.go:60]: You could make that more obvious by writing `... the namespace of the Catalog CR this ...`

**Comment by marians** [comment]
> Should we add catalogNamespace to the example CR in the UT?

Always good to have the example as complete as possible

**Review by marians — APPROVED** [review]

**Pull request merged**